eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenArtifactInfo;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenFormatStage;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenResolvedArtifact;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenResolverSystemBase;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenStrategyStage;
+import org.jboss.shrinkwrap.resolver.api.maven.PomEquippedResolveStage;
+import org.jboss.shrinkwrap.resolver.api.maven.PomlessResolveStage;
+import org.jboss.shrinkwrap.resolver.api.maven.ScopeType;
+import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenDependency;
+
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Utils {
+    public static final String COMMANDLINE_INPUTFILE = "inputfile";
+    public static final String COMMANDLINE_OUTPUTFILE = "outputfile";
+    public static final String COMMANDLINE_POMFILE = "pomfile";
+    public static final String COMMANDLINE_EXTERNALDEPENDENCIES = "externalDependencies";
+    public static final String COMMANDLINE_GROUPID = "groupid";
+    public static final String COMMANDLINE_EXCLUSIONLIST = "exclusionList";
+    public  static final Pattern COMMANDLINE_REGEX = Pattern.compile("-(.*)=(.*)");
+    public static final List<String> EXCLUSION_LIST = Arrays.asList("azure-spring-data-cosmos", "azure-spring-data-cosmos-test");
+    public static final Pattern SDK_DEPENDENCY_PATTERN = Pattern.compile("(.+):(.+);(.+);(.+)");
+    public static final Pattern EXTERNAL_DEPENDENCY_PATTERN = Pattern.compile("(.+):(.+);(.+)");
+    public static final Pattern SDK_NON_GA_PATTERN = Pattern.compile("(.+)-(.+)");
+    public static final String AZURE_CORE_GROUPID = "com.azure";
+    public static final String AZURE_TEST_LIBRARY_IDENTIFIER = "-test";
+    public static final String AZURE_PERF_LIBRARY_IDENTIFIER = "-perf";
+    public static final String AZURE_CORE_TEST_LIBRARY = "azure-core-test";
+
+    public static final String CONFLICTING_DEPENDENCIES = "conflict";
+    public static final String BOM_ELIGIBLE = "bom";
+    public static final HashSet<String> EXTERNAL_BOM_DEPENDENCIES = new HashSet<String>(Arrays.asList(
+        "io.projectreactor",
+        "com.fasterxml.jackson",
+        "io.netty",
+        "io.projectreactor.netty"
+    ));
+
+    public static final HashSet<String> RESOLVED_EXCLUSION_LIST = new HashSet<>(Arrays.asList(
+       "junit-jupiter-api"
+    ));
+
+    public static final String POM_TYPE = "pom";
+
+
+    public static List<BomDependency> getPomFileContent(Dependency dependency) {
```

Comment:
Given that we are using Maven dependencies in this project already, is there a library which helps with downloading a Maven POM file's contents and stores it in the users .m2, and if the POM file is already in the .m2 it uses that? Basically, using the same logic that Maven commands use.

Reply:
I actually searched for that and spend a good amount of time finding whether srinkwrap or any other library can do it. But couldn'tt find anything. In fact this ask was made on shrinkwrap since 2014 but hasn't seen any traction.

Reply:
If you know of something - happy to use it!